llvm/lib/CodeGen/GlobalISel/LoadStoreOpt.cpp

```diff
@@ -117,8 +117,12 @@ bool GISelAddressing::aliasIsKnownForLoadStore(const MachineInstr &MI1,
   if (!BasePtr0.BaseReg.isValid() || !BasePtr1.BaseReg.isValid())
     return false;
 
-  int64_t Size1 = LdSt1->getMemSize();
-  int64_t Size2 = LdSt2->getMemSize();
+  LocationSize Size1 = LdSt1->getMemSize() != MemoryLocation::UnknownSize
+                           ? LdSt1->getMemSize()
+                           : LocationSize::beforeOrAfterPointer();
+  LocationSize Size2 = LdSt2->getMemSize() != MemoryLocation::UnknownSize
+                           ? LdSt2->getMemSize()
+                           : LocationSize::beforeOrAfterPointer();
```

Comment:
How about we add helpers to `GLoadStore` like `getMemLocationSize()`. I find having to compare against this awkwardly named `beforeOrAfterPointer()` thing cumbersome.

Reply:
Thanks - The plan is for getMemSize() to become a LocationSize in #70452, so this should be cleaned up to a simpler `LocationSize Size1 = LdSt1->getMemSize();`. It just has to be a bit awkward for the moment, and eventually getMemSize will never be MemoryLocation::UnknownSize.

Reply:
Ok go ahead then.